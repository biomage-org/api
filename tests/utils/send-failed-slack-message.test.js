const sendFailedSlackMessage = require('../../src/utils/send-failed-slack-message');
const { USER } = require('../test-utils/constants');

const experiment = {
  meta: {
    pipeline: {
      stateMachineArn: 'qcArn',
    },
    gem2s: {
      stateMachineArn: 'gem2sArn',
    },
  },
};
global.fetch = jest.fn(() => Promise.resolve({ ok: true }));
describe('sendFailedSlackMessage', () => {
  beforeEach(() => {
    fetch.mockClear();
  });

  it('Sends slack message with correct fields for QC', async () => {
    const message = {
      input: {
        processName: 'qc',
        taskName: 'Task 1',
        authJWT: 'lmao',
      },
      experimentId: 'asd123',
    };

    await sendFailedSlackMessage(message, USER, experiment);
    expect(fetch.mock.calls).toMatchSnapshot();
  });

  it('Sends slack message with correct fields for Gem2s', async () => {
    const message = {
      input: {
        processName: 'gem2s',
        taskName: 'someStep',
        authJWT: 'lmao',
      },
      experimentId: 'anotherExpId',
    };
    await sendFailedSlackMessage(message, USER, experiment);
    expect(fetch.mock.calls).toMatchSnapshot();
  });
});
