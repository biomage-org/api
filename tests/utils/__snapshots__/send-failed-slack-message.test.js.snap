// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sendFailedSlackMessage Sends slack message with correct fields for Gem2s 1`] = `
Array [
  Array [
    "http://mockSlackHook.com/asdasd",
    Object {
      "body": "{\\"blocks\\":[{\\"type\\":\\"section\\",\\"text\\":{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"\\\\n          *gem2s* has failed for experiment *anotherExpId*\\\\n          *Step*: someStep\\\\n          *State machine ARN*: gem2sArn\\\\n          *Deployment*: local-test\\\\n          \\"}},{\\"type\\":\\"context\\",\\"elements\\":[{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User email:*\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"johncena@gmail.com\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User name:*\\"},{\\"type\\":\\"plain_text\\",\\"text\\":\\"John Cena\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User UUID:*\\"},{\\"type\\":\\"plain_text\\",\\"text\\":\\"032cdb44-0cd3-4d58-af21-850kp0b95ac7\\"}]}]}",
      "method": "POST",
    },
  ],
]
`;

exports[`sendFailedSlackMessage Sends slack message with correct fields for QC 1`] = `
Array [
  Array [
    "http://mockSlackHook.com/asdasd",
    Object {
      "body": "{\\"blocks\\":[{\\"type\\":\\"section\\",\\"text\\":{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"\\\\n          *qc* has failed for experiment *asd123*\\\\n          *Step*: Task 1\\\\n          *State machine ARN*: qcArn\\\\n          *Deployment*: local-test\\\\n          \\"}},{\\"type\\":\\"context\\",\\"elements\\":[{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User email:*\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"johncena@gmail.com\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User name:*\\"},{\\"type\\":\\"plain_text\\",\\"text\\":\\"John Cena\\"},{\\"type\\":\\"mrkdwn\\",\\"text\\":\\"*User UUID:*\\"},{\\"type\\":\\"plain_text\\",\\"text\\":\\"032cdb44-0cd3-4d58-af21-850kp0b95ac7\\"}]}]}",
      "method": "POST",
    },
  ],
]
`;
